### DISCUSSION TITLE
Web3Auth equivalent to Torus getTorusKey

### DISCUSSION THREAD

**Discussion by NTonani** [Web3Auth/Web3Auth#423] [Help]
We integrated with Torus last year using a custom JWT verifier. The UX is handled in-house and we provide a valid JWT downstream to `getTorusKey` for key material. 

The latest rebrand and SDK does not seem as willing to give clients the same flexibility, or at least we aren't finding it in the docs.

Tried following [BYO JWT Providers](https://web3auth.io/docs/custom-authentication/byo-jwt-providers) from the docs, two discoveries:
1. The latest [ExtraLoginOptions](https://web3auth.io/docs/sdk/web/openlogin#extraloginoptions) spec does not seem to support directly plumbing `id_token` through as the doc suggests doing.
2. We still need to initialize our client with an `OpenLoginAdapter`, which has a required `uxMode` field. We were hoping this was ignored when `connectTo` is called with `id_token` present, but that was not the case.

Can you explain the two discrepancies please? We are having troubles there.

And more generally, in the new Adapter/Provider model, is there an equivalent flow to directly calling `getTorusKey` on deprecated Torus client?

**Comment by YZhenY** [comment]
I'm in transit but will answer this as soon as i get on a proper station

**Comment by himanshuchawla009** [comment]
Hello, 
1. You can still pass `id_token` in extraLoginOptions.Seems like some typo in docs, we will fix the doc.

2. Unlike custom auth SDK `web3auth` doesn't have a `getTorusKey` implementation, even if you are passing id_token in options it will redirect to `open login` to complete the MFA process. We are working on something similar, we will update you once it is implemented.
